### PR TITLE
feat: add code coverage action

### PR DESCRIPTION
**Pull request by tim-romel-kr**
JIRA: [DCPEXP-80391](https://jira.kroger.com/jira/browse/DCPEXP-80391)

### CODE REVIEW

File: .github/actions/code-coverage/action.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: Code Coverage Report
+description: 'Adds a comment to PRs with code coverage'
+
+inputs:
+  coverage-counter-type:
```
Comment (ccrowell-kr) [inline .github/actions/code-coverage/action.yml:5]: This breaks our convention of underscores for properties. Looks like a mixup of the kover action property and ours - from below: `coverage-counter-type: ${{ inputs.coverage_counter_type }}`

File: .github/actions/code-coverage/action.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    description: 'Report counter type used to calculate coverage metrics. Possible values are: INSTRUCTION, LINE or BRANCH.'
+    required: true
+    default: 'LINE'
+  gradle_command:
+    description: 'The gradle command to execute to generate the code coverage report'
+    required: true
+    default: 'koverXmlReport'
```
Comment (ccrowell-kr) [inline .github/actions/code-coverage/action.yml:12]: Do defaults and required together make sense?

File: .github/actions/code-coverage/action.yml (hunk 1 above)
Comment (tim-romel-kr) [reply .github/actions/code-coverage/action.yml]: Yes I was originally copy/pasting from the action we are using until I realized the conventions don't match. Must have missed switching one. Good to know an action does not fail when referencing an undeclared input  😬 .

File: .github/actions/code-coverage/action.yml (hunk 2 above)
Comment (tim-romel-kr) [reply .github/actions/code-coverage/action.yml:10]: It made sense at the time 😅 . Thought was that having required set to true even with a default given would not allow a null or empty value to be supplied. I went ahead and removed required wherever a default is given.

**Review by ccrowell-kr — APPROVED** [review]

**Pull request merged**